llzk/src/symbol_ref.rs

```diff
@@ -100,3 +100,10 @@ impl<'c> fmt::Display for SymbolRefAttribute<'c> {
         write!(f, "{}", self.inner)
     }
 }
+
+/// Equivalent to `SymbolRefAttr` in MLIR, providing a common trait for
+/// both flat and non-flat symbol reference attributes.
+pub trait SymbolRefAttrLike<'c>: AttributeLike<'c> {}
+
+impl<'c> SymbolRefAttrLike<'c> for SymbolRefAttribute<'c> {}
+impl<'c> SymbolRefAttrLike<'c> for FlatSymbolRefAttribute<'c> {}
```

Comment:
Do we want to allow crates downstream to implement this trait? We need to [seal it](https://rust-lang.github.io/api-guidelines/future-proofing.html#sealed-traits-protect-against-downstream-implementations-c-sealed) otherwise

Reply:
I think I tried creating another subclass of `SymbolRefAttr` in `llzk-lib` at one point and it wasn't possible so I suppose we should seal it here.